### PR TITLE
Profile command should not create non existent profiles

### LINKED ISSUE #6647: Prevent minikube profile command to create profile
one un-intentional side effect of minikube profile command is creating a new profile.

minikube profiles should only be created by minikube start.

so if you by mistake do

```
minikube profile lis
```

(instead of list)

it will create an abandoned profile called list.
I would like to disable the creation of profile by minikube profile command.

however, 'minikube profile pname'
should still change the profile to 'pname' if it exists. but should not create a new one.

and it shoud also send a nice message to the user if it doesnt exist.

"The profie lis doesnt exist, if you want to create a profile you can by this command
minikube start -p lis
"

### PR DESCRIPTION
**Pull request by aallbrig**
### The Change Set
Fixes #6647

This updates `profile` to no longer create profiles that do not exist and provides feedback to user on how to create the non-existent profile.

### Just One Snag...

One thing brought up in the issue that doesn't appear to be true is...

> however, 'minikube profile pname'
> should still change the profile to 'pname' if it exists. but should not create a new one.

However; **`minikube profile pname` does not change `profile` to `pname` before this change set**. This seems like it would be **new functionality**

``` bash
$ minikube create -p profile
$ minikube profile pname
```

Instead of _renaming `profile` to `pname`_, it instead creates a `pname` profile that needs to be deleted.

TLDR; This aspect of issue #6647 may be a new feature request because this behavior does not exist today.

See below screenshots.

Screenshot 2020-02-17 18 53 24
Screenshot 2020-02-17 18 53 33
Screenshot 2020-02-17 18 53 43
Screenshot 2020-02-17 18 53 54
Screenshot 2020-02-17 18 54 01
Screenshot 2020-02-17 19 04 47

### CODE REVIEW

**Comment by medyagh** [comment]
Thank you for the pr. Do you mind adding integration tests for this ?

You could look up examples of functional tests and add one more tests under functional tests

**Comment by aallbrig** [comment]
@medyagh I can but not tonight. I’m spending time with the fiancé for the rest of the night.

It’s 9:14 p.m. EST. I’ll see about getting this done between 7 a.m. and 8 a.m. tomorrow.

**Comment by aallbrig** [comment]
@medyagh I'm on it now

**Comment by aallbrig** [comment]
@medyagh I found that there were integration tests for profile already existing! I created tests for nonexistent profiles and the desired behavior described in the issue.

I look forward to your review. Its 7:12 a.m. EST now. I'll have a block of time during lunch and another block of time after work to address feedback you (or others) provide.

**Comment by aallbrig** [comment]
@medyagh Is there anything else I can do?

**Comment by medyagh** [comment]
@aallbrig thank you so much for this PR ! this looks good to me. and thank you for adding integraiotn tests for it too

File: test/integration/functional_test.go [hunk 1]
```diff
@@ -494,7 +494,25 @@ func validateLogsCmd(ctx context.Context, t *testing.T, profile string) {
...
 // validateProfileCmd asserts "profile" command functionality
 func validateProfileCmd(ctx context.Context, t *testing.T, profile string) {
-	rr, err := Run(t, exec.CommandContext(ctx, Target(), "profile", "list"))
+	// Profile command should not create a nonexistent profile
+	nonexistentProfile := "lis"
+	rr, err := Run(t, exec.CommandContext(ctx, Target(), "profile", nonexistentProfile))
+	if err != nil {
+		t.Errorf("%s failed: %v", rr.Args, err)
+	}
+	for _, line := range []string{fmt.Sprintf("Created a new profile : %s", nonexistentProfile), fmt.Sprintf("minikube profile was successfully set to %s", nonexistentProfile)} {
```
Comment (medyagh) [inline test/integration/functional_test.go]: how about instead of parsing the output, do a 
"minikube profile list --output=json" and expect not to see the "lis" in the created profiles?

Comment (aallbrig) [reply test/integration/functional_test.go]: Absolutely! I’ll update it now

**Comment by aallbrig** [comment]
@medyagh Touching base: doing a trivia night with friends & S.Os. My plan is to do the refactor later today. Cheers!

**Comment by medyagh** [comment]
> trivia night with friends & S.Os. My plan is to do the refactor later today. Cheers!

thanks for keeping me up to date ! have fun at the trivia :)

**Comment by aallbrig** [comment]
https://youtu.be/_TANF069X40

**Comment by aallbrig** [comment]
@medyagh Updated tests to use JSON output to determine expected behavior. Please let me know how I can improve this change set :)

**Comment by aallbrig** [comment]
Woohoo!
